### DISCUSSION TITLE
Design codes

### DISCUSSION THREAD

**Discussion by colmjude** [digital-land/data-standards-backlog#27] [Planning consideration]
# Design codes

**Comment by RichLimbrick** [comment]
Hi @colmjude - just looking at the [design codes standard](https://digital-land.github.io/specification/specification/design-code/) to see if it can meet a need for a PlanX flow for a householder application or prior approval in an area of the borough where we have quite narrow accepted design requirements - albeit there is no formal design code, just a consistent approach. I wondered if you had pre-defined values for `design-code-status` or `design-code-categories` (sorry if I missed them). A code with an "informal" status (or similar) could allow us to code these types of accepted approaches that don't fall within a formal design code:
`design-code-status` — a [design-code-status](https://digital-land.github.io/specification/specification/design-code/#design-code-status) reference
`design-code-categories` — a list of one or more [design-code-category](https://digital-land.github.io/specification/specification/design-code/#design-code-category) references, separated by a semi-colon ';' character

**Reply by mattlucht** [reply]
Hey @RichLimbrick the design code standard is very much in draft (we were discussing with @psd yesterday about how we may revisit this soon). It'd be good to see whether we can improve this spec to help meet your needs.

That said, we do have some initial [design-code-status](https://github.com/digital-land/design-code-collection/blob/main/data/design-code-status.csv) and [catergories](https://github.com/digital-land/design-code-collection/blob/main/data/design-code-category.csv). I'm not sure the current statues meet your needs, we have `proposed`, `draft`, `adopted` and `withdrawn`. 

Happy to discuss this!

**Reply by mattlucht** [reply]
Oh, and the lists of status and categories are definitely hard to find at the moment, but we're looking at how we can more easily surface or link to them in the specification pages!

**Reply by FreyaLBS** [reply]
Hi @mattlucht, would you please be able to give an update on if you were able to find out more about this? At Southwark Council we are exploring design codes within our planning team and looking to explore more of what we can achieve with PlanX. Thanks in advance

**Reply by RichLimbrick** [reply]
This is a really good point on the support for PlanX. If we can get an "Informal" status this could help us flag any emerging design guidance and design codes to applicants really early in the process. It would put design codes out in front of agents and applicants. Happy to test this one out for you in due course.

**Reply by mattlucht** [reply]
Hi @FreyaLBS @RichLimbrick – this sounds great and happy to work with you to shape something that works for you and adds value for PlanX (and beyond!).

Although the specification is still in draft it'd be great to start testing this to see how it all works. I've added an additional status of `informal` – it's missing a description but I can borrow from your words above to help fill this out! 

https://dluhc-datasets.planning-data.dev/dataset/design-code-status

**Comment by colmjude** [comment]
## Understanding design codes

Design codes are becoming a higher priority for us because of the need for all LPAs to soon produce one covering their district.

It has been a while since we looked at Design Codes, so we thought we’d start by refreshing our understanding of Design Codes.

We thought it’d be useful to summarise our understanding to see if we are getting it right.

We noticed design code is used to mean 2 things:

**Design Code** = a set or collection of specific design rules and criteria for development in a broad area (usually a local authority district). For example, [Trafford Design Code](https://trafforddesigncode.uk/)
**design code** (or code) =  a specific rule/requirement and related documentation contained within a Design Code. For example, [SPSD5 On-street car parking](https://trafforddesigncode.uk/streets-and-public-realm/street-design/)

A Design Code is a set of design requirements providing specific parameters for the physical development of a site or area. They set expectations of design quality for any upcoming development.

Each design code should be more visual and set clearer, more measurable parameters to work to than other types of guidance.

There are 10 different themes that a design code could sit under.
Each theme has a set of design aspects or qualities a design code can cover.
For example, Built Form is a theme; within that, a design code could cover density, building heights or building lines.

Each design code can be scoped (or classified) by various criteria. This can be done by

- Location (area type e.g. all suburbs, named places e.g. town centre, or development site e.g. development of 1000+ houses)
- Type of application (for example, homeowner extensions)
- Development pressures or issues (for example, tall buildings)
- Theme (for example, Built form or Movement)

Local planning authorities will have to create a Design Code containing a set of design codes.
Design Codes should be created in addition to Local Plans. Initially, Design Codes are classed as Supplementary Planning Documents (SPDs) but may be Supplementary Plans (SPs) in the future.

A Design Code is produced in a similar way to other policy instruments (e.g. Local Plans). They require evidence gathering, stakeholder engagement, drafting and testing, consultation and adoption.

Entities other than local planning authorities can or should create Design Codes. They are:

- applicants of major developments (100s of homes)
- Landowners have the option
- Neighbourhood plan forums

### Questions

How do local plans and design codes intersect?
If a design code is applicable to a named place, such as the town centre, are the bounds of that place defined?
Who defines or sets the development pressures or issues? Are these unique to an area?
What is classed as a major development for which a design code is required? 
What are the criteria by which a Landowner can draw up their own design code?

If you are able to help us answer any of these questions, please shout.

### Useful links

- [The National Model Design Code Part 1](https://www.gov.uk/government/publications/national-model-design-code/national-model-design-code-part-1-the-coding-process-html-accessible-version)
- [The National Model Design Code Part 2](https://assets.publishing.service.gov.uk/media/611105f98fa8f506c58e786f/National_Model_Design_Code_-_Part_2_Guidance_Notes_web.pdf)
- [National design guide](https://www.gov.uk/government/publications/national-design-guide/national-design-guide-accessible-version)

**Comment by colmjude** [comment]
## User needs

We have been exploring user needs around design codes. User needs help us create effective data standards so the better we understand the needs around the data the better the standard should be.

This is by no means an exhaustive list but here are a few we have come up with. We need to verify whether these are right. We've tried to look at a few different users so there might be some repetition. Different users have similar needs but we felt it was useful to look at design codes through these different lenses.

### Developer

_High-level need_
As a developer I need to know if there is a design code I need to adhere to so that my application has the best chance of being granted

_Sub-needs_
- As a developer I need to know if a site is within an area covered by a design code so that I can prepare my application accordingly
- As a developer I need to know how to find out if there is a design code covering a given location so that I can prepare my application accordingly
- As a developer I need to know what area or areas a design code applies to so that I know what design codes I need to consider during my search for a site
- As a developer I need to know what area types a code applies to so that I can work out if design codes are applicable to the site
-- As a developer I need to know the area type of the area my site is in so that I can work out if I need to adhere to a design code 
- As a developer I need to know what design aspects a design code covers so that I can prepare my plans accordingly
- As a developer I need to know what application types a design code applies to so that I know if it is applicable to my application

_High-level need_
As a developer I need to understand how to meet the design code so that I can design my development and prepare my application

_High-level need_
As a developer, I need to be able to maintain a consistent character and cohesive delivery of a large site by multiple developers so that I can sell off parcels of land. 
_Design codes are a solution to this need._

### Design code author/creator

_High-level need_
As a design code author I need to know how to create design codes so that I have effective design codes covering an area

_Sub-needs_
- As a design code author I need to know what must be included in a design code so that it can be effective
- As a design code author I need to know what design codes have been created/granted so that I can refer to good design codes when I am creating mine
- As a design code author I need to find similar design codes to the one I want to create so that I can see how others have prepared theirs
- As a design code author I need to know who authored a design code so that I can speak to them to find out more information
- As a design code author I need to know what types of codes I can create and apply to an area so that I can create suitable codes for my jurisdiction
- As a design code author I need to clearly define the area types a code applies to so that I can be confident applicants will know what area type their site is in

### Local planning authority

_High-level need_
As a local planning authority I need to know what we need to do so that we are compliant with the design code policy

_Sub-needs_
- As a local planning authority we need to know how to create a design code
- As a local planning authority we need to know if any design codes exist in our area so that we know if we (the LPA) need to create one
- As a local planning authority we need to know if any part of our local authority area is not covered by a design code so that we can be confident we are satisfying the policy requirement

_High-level need_
As a local planning authority we need to be able to require a design code to be create as a condition of outline permission so that we can be confident the whole site will have a consistent character and can be delivered cohesively by multiple developers

### Other

_Sub-need_
As a design code user I need to know what the different area types are so that I can determine where design codes apply

We are interested to know if you think there are any we’ve captured that you disagree with or if there are any critical needs you think we’ve missed.

**Comment by mattlucht** [comment]
We've been working with the team at Office for Place to look at how the draft design code standard might be iterated. We've drafted some changes on this [Mural board](https://app.mural.co/t/mhclg2837/m/mhclg2837/1709140526813/9a28b5a76405c33dbb1568a2870a5e6057bf0b91?sender=u254215e050d2f98c69580938) which is open for discussion. 

Next we'd like to look at some existing design codes and try to recreate them to this data model.

<img width="1413" alt="CleanShot 2024-02-29 at 14 25 00@2x" src="https://github.com/digital-land/data-standards-backlog/assets/293065/d2ef2a7c-c724-4efb-a6ad-1926f3c52c05">

**Reply by colmjude** [reply]
Our work with the Office for Place on seeding some design code data has given us more confidence that these iterations to the data model are worth making.

We've now made these changes. The [technical specification](https://digital-land.github.io/specification/specification/design-code/) and field descriptions have been updated.

**Comment by colmjude** [comment]
We have been assessing the need for each field in the [design code specification](https://digital-land.github.io/specification/specification/design-code) and there are a couple we are not sure are needed (for now).

The main ones are 
* `document-url` in the [design-code-rule dataset](https://digital-land.github.io/specification/specification/design-code/#design-code-rule-dataset)
* `document-url` in the [design-code-area dataset](https://digital-land.github.io/specification/specification/design-code/#design-code-area-dataset)

We follow a pattern where [if a dataset is asking for a `document-url` we always pair it with a `documentation-url`](https://standards.planning-data.dev/rules/#a-record-with-a-document-url-should-also-have-a-documentation-url). Documents, particularly pdfs have a habit of disappearing. That’s why we ask for the `documentation-url`, which should include information about the publication. This means if the document disappears we know where we should look for updates or new `document-urls`.

With design codes there won’t be individual “documents” for a design code rule or design code area. 

Design code rules and areas are defined and contained within a Design Code (the overarching publication). This Design Code needs both a `document-url` and a `documentation-url` — the Design Code is a body of work that is published somewhere (website or pdf), and there should be a page or section on the planning authority site that references or provides details about the publication. In this case, `document-url` doesn't add much.

The other field we are questioning is `design-code` in the [`design-code-area` dataset](https://digital-land.github.io/specification/specification/design-code/#design-code-area-dataset). We are wondering if this can this be inferred? Design code rules apply to an area and these rules are defined in a Design Code. If so, we might not need this field. We need to discuss this further.

**Reply by colmjude** [reply]
For the reason outlined above - how we only need `document-url` AND `documentation-url` when we are expecting data about a document. And, coupled with our principle of starting small and evolving a data model from there, we’ve decided to remove  `document-url` field from the `design-code-rule` and `design-code-area` datasets. 

If we discover a compelling reason to include them and believe it will be feasible for providers to generate that data we will look to add them back in.

[v1.3.3 of design code specification](https://digital-land.github.io/specification/specification/design-code/v1.3.3/)

**Reply by colmjude** [reply]
We've decided to re-instate the `document-url` field on design-code-rule because it allows the data to deep link directly to the rule if that is possible.

For example, with the [Healthy Streets for Surrey Design Code](https://healthystreets.surreycc.gov.uk/) it is possible to link directly to rules.
In this case the Street vision and strategy rule would have 
* the `document-url` = [https://healthystreets.surreycc.gov.uk/requirements-and-guidance/section?id=4.1](https://healthystreets.surreycc.gov.uk/requirements-and-guidance/section?id=4.1) - the rule itself
* the `documentation-url` = [https://healthystreets.surreycc.gov.uk/requirements-and-guidance/section?id=4](https://healthystreets.surreycc.gov.uk/requirements-and-guidance/section?id=4) - the page linking to the specific rule

[Spec v1.4.4](https://digital-land.github.io/specification/specification/design-code/changelog)

**Comment by colmjude** [comment]
### Design Code Areas

Capturing design code areas is vital to the value of this data. It unlocks how users can access Design Codes and the rules/requirements.

Instead of finding a Design Code for a large area then having to interpret it themselves users would be able to see what rules/requirements apply to the area they care about or want to develop on a much more specific level. For example, these 12 rules matter on this street I want to develop on instead of these 179 rules *could* apply to this street but let me read the whole Design Code  to work that out.

During our recent work to seed some data to test the data mode with, we noticed that areas are defined in a variety of ways. In some cases it was easy enough to find precise areas (although the data wasn't readily available), in other cases the areas were rough approximations.

Here are some examples of what we found

![lye-design-code-all-areas](https://github.com/digital-land/data-standards-backlog/assets/299875/9031bbb7-65ae-4c67-a2aa-a138566d23e2)
The [Lye and Stour design code](https://www.bdp.com/globalassets/--misc--/lye-masterplan-and-design-code/design-code_draft.pdf) had clearly defined areas in their pdf publication.

![trafford_design_code_areas](https://github.com/digital-land/data-standards-backlog/assets/299875/f9ef3df9-872d-4274-9ede-3c6f7521d2d6)
[Trafford design code](https://trafforddesigncode.uk/wp-content/uploads/2023/06/Trafford-Design-Code_text-only-version_June2023.pdf) had something similar. But the types don't quite match with the defined area types in [design-code-area-types dataset](https://dluhc-datasets.planning-data.dev/dataset/design-code-area-type).

![uttlesford-design-code-area-types](https://github.com/digital-land/data-standards-backlog/assets/299875/4097e68a-d512-4747-b3de-4006743f5e23)
The [Uttlesford design code](https://www.essexdesignguide.co.uk/uttlesford-design-code/uttlesford-places/settlement-types/) has rough regions on a map. It would be difficult to provide that precision that rules X, Y and Z apply to an exact location.
